scenes/enumerated/problem_10349/problem_10349.py

```diff
@@ -0,0 +1,393 @@
+        question_mark_2 = MathTex(question_mark_string).next_to(one_part_copy, UP, buff=.2)
+        self.play(FadeIn(question_mark_1), FadeIn(question_mark_2))
+        self.wait()
+        self.play(FadeOut(question_mark_1), FadeOut(question_mark_2))
+        self.wait()
+
+        # Հեռցնենք պայմանները
+        checks_and_conditions = [[FadeOut(check_marks[i]), FadeOut(conditions[i])] for i in range(len(conditions))]
+        self.play(AnimationGroup(
+            *flatten_list(checks_and_conditions),
+            lag_ratio=0.3,
+            run_time=1
+        ))
+
+        # Դեղինով շրջապտույտ նդգծենք 10֊երի շուրջ
+        self.play(*[Circumscribe(x3_one_part_and_10[i][1], fade_out=True) for i in range(3)])
+        road_new = VGroup(*[x3_one_part_and_10[i] for i in reversed(range(3))], *[x5_one_part[i] for i in range(5)])
+        self.wait()
+        self.play(
+            FadeOut(
+                road, souren_road, arshak_road,
+                souren_road_time, arshak_road_time,
+                road_length, souren_t, arshak_t,
+                souren_road_brace, arshak_road_brace
+            ),
+            road_new.animate.align_to(souren_t, DOWN)
+        )
+        self.remove(road, souren_road, arshak_road)
+        self.wait()
+
+        # Հաշվենք 1-ից մինչև 8 մասերը
+        count_x8_one_part_labels = [
+            *[Tex(str(i + 1)).next_to(x5_one_part[i], UP) for i in range(5)],
+            *[Tex(str(8 - i)).next_to(x3_one_part_and_10[i][0], UP) for i in range(3)]
+        ]
+        count_x8_one_part_animation = AnimationGroup(
+            *[Write(count_x8_one_part_labels[i]) for i in range(5)],
+            *[Write(count_x8_one_part_labels[i + 5]) for i in reversed(range(3))],
+            lag_ratio=.5, run_time=2
+        )
+        self.play(count_x8_one_part_animation)
+        self.play(
+            *[FadeOut(count_x8_one_part_labels[i]) for i in range(8)]
+        )
+
+        # Մկրատով կտրենք 10-երը
+        scissors = [
+            *[DScissors(x3_one_part_and_10[i][1].left_edge.get_center())
+              for i in range(len(x3_one_part_and_10))],
+            *[DScissors(x3_one_part_and_10[len(x3_one_part_and_10) - 1 - i][1].right_edge.get_center())
+              for i in range(len(x3_one_part_and_10) - 1)]
+        ]
+        self.play(*[CutIn(scissors[i]) for i in range(len(scissors) - 2)])
+        self.play(*[CutOut(scissors[i]) for i in range(len(scissors) - 2)])
+        self.play(*[CutIn(scissors[i + 3]) for i in range(len(scissors) - 3)])
+        self.play(*[CutOut(scissors[i + 3]) for i in range(len(scissors) - 3)])
+
+        for i in range(len(x3_one_part_and_10)):
+            x3_one_part_and_10[i][1].set_label(Tex(segment_10_string))
+
+        segment_10_labels = [segment_10_label.copy() for _ in range(len(x3_one_part_and_10))]
+
+        self.play(AnimationGroup(
+            *[x3_one_part_and_10[i][1].animate.shift(UP * .5) for i in range(len(x3_one_part_and_10))],
```

Comment:
Iterate through objects, not indices
```suggestion
            *[part[1].animate.shift(UP * .5) for part in x3_one_part_and_10],
```